frontend/appflowy_flutter/lib/workspace/presentation/home/home_layout.dart

```diff
@@ -19,26 +19,42 @@ class HomeLayout {
   late double homePageROffset;
   late double menuSpacing;
   late Duration animDuration;
-
+  
   HomeLayout(BuildContext context, BoxConstraints homeScreenConstraint) {
     final homeSetting = context.read<HomeSettingBloc>().state;
 
     showEditPanel = homeSetting.panelContext.isSome();
-
+    
     menuWidth = Sizes.sideBarMed;
     if (context.widthPx >= PageBreaks.desktop) {
       menuWidth = Sizes.sideBarLg;
     }
 
     menuWidth += homeSetting.resizeOffset;
 
-    if (homeSetting.isMenuCollapsed) {
-      showMenu = false;
-    } else {
-      showMenu = true;
-      menuIsDrawer = context.widthPx <= PageBreaks.tabletPortrait;
-    }
+    bool isSmallScreen = false;
+
+    if (context.widthPx < PageBreaks.tabletPortrait) isSmallScreen = true;
+    context
+        .read<HomeSettingBloc>()
+        .add(HomeSettingEvent.isSmallScreen(isSmallScreen));
 
+    if (isSmallScreen) {
+      if (homeSetting.pinSideBar) {
+        showMenu = true;
+        menuIsDrawer = context.widthPx <= PageBreaks.tabletPortrait;
+      } else {
+        showMenu = false;
+      }
+    } else {
+      if (homeSetting.isMenuCollapsed && !homeSetting.pinSideBar) {
+        showMenu = false;
+      } else {
+        showMenu = true;
+        menuIsDrawer = context.widthPx <= PageBreaks.tabletPortrait;
+      }
+    } 
```

Comment:
home_setting_bloc.dart tracks whether the navigation bar is expanded or collapsed, but the UI is doing the calculations. I think it would be more appropriate to move this logic to the BLoC and emit an event containing the screen width, given that the BlocBuilder only builds when `p.isMenuCollapsed != c.isMenuCollapsed || p.isSmallScreen != c.isSmallScreen`.

Reply:
Thanks @a-wallen  will review and revert

Reply:
@a-wallen please can you help me understand better what the emitted screen width will be used for?